### PR TITLE
License fixes (part 2)

### LINKED ISSUE #275: Simplified BSD License is referenced but not included
##### SUMMARY

Some module_utils and such use the Simplified BSD License but the file itself is not included in the collection. 
Partially addressed in #280, the rest in #282 .

See also https://github.com/ansible-collections/community.general/pull/4759 for example.

### PR DESCRIPTION
**Pull request by briantist**
##### SUMMARY

Related:
- #280 
- https://github.com/ansible-community/community-topics/issues/112
- Fixes #275 

This is part 2 of what's implemented in #280 . It fixes the headers in existing files, and adds SPDX headers.

It already includes those changes and will be rebased after #280 is merged.

The commit created by this PR being merged will be added to a new [`.git-blame-ignore-revs` file](https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view).

~After #280 is merged and this is rebased, I may try adding the ignore revs file in the same PR, and using the "Rebase and merge" option instead of "Squash and merge"; I _think_ that will preserve the original commit hash of the one I want to ignore, and put both commits in `main`. If it doesn't work right, I can always fix it after.~

[From the docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#rebase-and-merge-your-pull-request-commits):

> The rebase and merge behavior on GitHub deviates slightly from `git rebase`. Rebase and merge on GitHub will always update the committer information and create new commit SHAs, whereas `git rebase` outside of GitHub does not change the committer information when the rebase happens on top of an ancestor commit.

[And some more discussion about the feature on SO](https://stackoverflow.com/q/48350294/3905079).

##### ISSUE TYPE

- Bugfix Pull Request

### CODE REVIEW

File: changelogs/fragments/licensing.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+bugfixes:
+  - Use the correct GPL license for plugin_utils (https://github.com/ansible-collections/community.hashi_vault/issues/276).
+  - Add missing ``simplified_bsd.txt`` file for BSD licensed content (https://github.com/ansible-collections/community.hashi_vault/issues/275).
```
Comment (gotmax23) [inline changelogs/fragments/licensing.yml:4]: Nit: The actual PR has `LICENSES/BSD-2-Clause.txt`

Comment (briantist) [reply changelogs/fragments/licensing.yml]: this was just a lag in rebasing, it's updated now in https://github.com/ansible-collections/community.hashi_vault/pull/282/commits/126b9e6581e2055fd6e099cc721e365b67507ee5 :)

Comment (briantist) [reply changelogs/fragments/licensing.yml]: @gotmax23 should be all set now if you want to take a look

**Review by felixfontein — APPROVED** [review]

**Comment by briantist** [comment]
@felixfontein thanks again for reviewing, and for the script to put these headers in place

**Pull request merged**